samplestore/src/main/java/com/stripe/samplestore/PaymentActivity.java

```diff
@@ -174,7 +188,7 @@ public void onSuccess(@NonNull PaymentAuthResult result) {
 
                     @Override
                     public void onError(@NonNull Exception e) {
-                        finishPayment();
+                        displayError(e.getMessage());
```

Comment:
Is `onCompleted` called if an error happened? If not, I think it should be , because we need to restart the payment process (i.e. re-collect payment method)